### DISCUSSION TITLE
Getting 'Failed.' For all operation after updating BetterDisplay to Version 2.3.1 Build 29141

### DISCUSSION THREAD

**Discussion by elzii** [waydabber/betterdisplaycli#4] [General]
Title covers entiry of error. This command for instance, worked reliable for months, no longer work:

```sh
betterdisplaycli set --name="V655-J09" --brightness=100% --temperature=0% --gamma=0% --constrast=0% --gain=0%
```

**Comment by waydabber** [comment]
Hmm. Just checked the command with BD 2.3.1 (substituting a display of mine to the name) and it seems to work, the command should be fine.

The only way you should get an instant failed for this is if the display's name is wrong or color table based software image adjustments are not enabled for the display (with Metal or overlay based adjustments not all of these controls are available).

You might want to try to see which command fails. You can try for example with `betterdisplay set --namelike=v655 --brightness=100%` and see if it works, then add additional commands. You can also check under the display's Image Adjustments submenu what image controls are available for the display.

**Comment by elzii** [comment]
![ZIZZOCAPTURE--2024-04-30@02 13 31-PM](https://github.com/waydabber/betterdisplaycli/assets/1827950/53c307bb-3d1d-4a90-99ce-9ac6a13d3f7e)

Sanity checked versions, tried released from this repo instead of symlinking to app...

Tried to change to macOS built in display in the event my monitor name changed....

![ZIZZOCAPTURE--2024-04-30@02 14 40-PM](https://github.com/waydabber/betterdisplaycli/assets/1827950/94526a89-c1e9-4376-ad7f-b1f32e45b094)


Gave the bin direct accessibility access (though I don't think it even required it better)

Not working over via the REST api either. 


Out of ideas for the day 🤔

**Reply by waydabber** [reply]
Sorry, did not see this comment for some reason. Yeah, the syntax is wrong there. You should use it like this for example (with the latest app version):

`betterdisplaycli get -n=built -brightness`

This should return the brightness for the built-in display (or any display that has some string `built` in it).

**Comment by elzii** [comment]
@waydabber (I have Pro license if that matters) 
![ZIZZOCAPTURE--2024-04-30@02 19 14-PM](https://github.com/waydabber/betterdisplaycli/assets/1827950/3515ef3b-85db-400f-ba6e-4144fbd50ea0)

**Comment by waydabber** [comment]
Does anything work? Like simply showing the help?

But the examples above should _not_ work due to various wrong syntaxes (you omitted `-` or `--` from before `brightness` and the `=` after most of the filter parameters). Also if you use `name`, the name should be full. For `model` if you provide a hexadecimal number, you need to add a `0x` prefix to it.

Try:

`betterdisplaycli get --namelike=built --brightness`

**Comment by vaibhav3027** [comment]
even this
`betterdisplaycli proAvailable`
shows failed.

Every other command except `betterdisplaycli help` shows "Failed."

**Reply by waydabber** [reply]
`betterdisplaycli get -proAvailable`

You always need some initial command like `get`, `set`, `toggle` etc. All other parameters must start with a `-` or `--` (depending on the unix flavor you like).
